paasta_tools/adhoc_tools.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2015-2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+
+import service_configuration_lib
+
+from paasta_tools.long_running_service_tools import LongRunningServiceConfig
+from paasta_tools.utils import deep_merge_dictionaries
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import get_paasta_branch
+from paasta_tools.utils import load_v2_deployments_json
+from paasta_tools.utils import NoConfigurationForServiceError
+from paasta_tools.utils import NoDeploymentsAvailable
+from paasta_tools.utils import prompt_pick_one
+
+
+log = logging.getLogger(__name__)
+
+
+def load_adhoc_job_config(service, instance, cluster, load_deployments=True, soa_dir=DEFAULT_SOA_DIR):
+    general_config = service_configuration_lib.read_service_configuration(
+        service,
+        soa_dir=soa_dir
+    )
+    adhoc_conf_file = "adhoc-%s" % cluster
+    log.info("Reading adhoc configuration file: %s.yaml", adhoc_conf_file)
+    instance_configs = service_configuration_lib.read_extra_service_information(
```

Comment:
Plural since it loads every instance config in a file.

I copied this straight from `marathon_tools`.